ppa/archive/management/commands/generate_textcorpus.py

```diff
@@ -0,0 +1,262 @@
+        "work_type": "work_type_s",
+        "source": "source_t",
+        "source_url": "source_url",
+        "sort_title": "sort_title",
+        "subtitle": "subtitle",
+    }
+
+    # Argument parsing
+    def add_arguments(self, parser):
+        """
+        Build CLI arguments for command.
+        """
+
+        # add --path argument for output storage
+        parser.add_argument(
+            "--path",
+            help="Directory path to save corpus file(s)."
+            "Defaults to ./ppa_corpus_{timestamp}",
+            default=None,
+        )
+
+        # add --doc-limit argument (determines how many results to retrieve from solr)
+        parser.add_argument(
+            "--doc-limit",
+            type=int,
+            default=0,
+            help="Limit on the number of documents for corpus generation."
+            "The default of 0 considers ALL documents.",
+        )
+
+        # add --batch-size argument (for solr querying)
+        parser.add_argument(
+            "--batch-size",
+            type=int,
+            default=DEFAULT_BATCH_SIZE,
+            help="Number of docs to query from solr at one time",
+        )
+
+        # add --dry-run argument (don't save anything, just iterate)
+        parser.add_argument(
+            "--no-gzip",
+            action="store_true",
+            help="Save uncompressed as ppa_pages.jsonl"
+            "instead of compressed as ppa_pages.jsonl.gz",
+        )
+
+        # add --dry-run argument (don't save anything, just iterate)
+        parser.add_argument(
+            "--dry-run",
+            action="store_true",
+            help="Do not save anything, just iterate over solr",
+        )
+
+    #### SOLR ####
+
+    def iter_solr(self, item_type="page"):
+        """
+        Iterate over solr documents of a certain `item_type`.
+        """
+
+        # get params
+        batch_size, lim = self.batch_size, self.doclimit
+
+        # search for this item type, order by id
+        qset = self.query_set.filter(item_type=item_type).order_by("id")
+
+        # if we're looking for pages, rename those fields
+        if item_type == "page":
+            qset = qset.only(**self.PAGE_FIELDLIST)
+        elif item_type == "work":
+            qset = qset.only(**self.WORK_FIELDLIST)
+
+        # get the total count for this query
+        total = qset.count()
+        if not total:
+            raise Exception("no records found in solr for query")
+
+        # if we want fewer than that, decrease "total"
+        # (this has the effect of pulling from solr only what we need,
+        # since we limit how many rows we pull by this amount)
+        if lim and total > lim:
+            total = lim
+        # define a generator to iterate solr with
+        batch_iterator = (
+            result
+            for step in range(0, total, batch_size)
+            for result in qset[
+                step : (step + batch_size) if (step + batch_size) <= total else total
```

Comment:
was this causing an error?

Reply:
yes! 😬 We needed to make sure the batch_size doesn't push us over the limit of what we want to pull from solr.

Reply:
is that a parasolr flaw?

Reply:
not really, since solr doesn't know otherwise what our requested limit is. Just a mistake in how I thought resetting the batch_size might work, which is what I did earlier